consensus/state.go

```diff
@@ -1470,7 +1470,23 @@ func (cs *ConsensusState) recordMetrics(height int64, block *types.Block) {
 			missingValidators++
 			missingValidatorsPower += val.VotingPower
 		}
+
+		for _, val := range cs.LastValidators.Validators {
+			privValAddress := cs.privValidator.GetPubKey().Address()
```

Comment:
this will panic if cs.privValidator is nil!

Reply:
we should use `cs.LastValidators.GetByAddress(cs.privValidator.GetPubKey().Address())` function to get validator address instead of `for` loop.